README.md

```diff
@@ -69,32 +69,103 @@ for example in data_loader:
+details insteadf.
+
+### <a name="Docker"></a>Docker
 
 The included Dockerfile creates a standalone demo image with
 Jupyter, Pytorch, Spark, and rikai preinstalled with notebooks for you
 to play with the capabilities of the rikai feature store.
 
-### Building the image
+To build and run the docker image from the current directory:
+```bash
+# Clone the repo
+git clone git@github.com:eto-ai/rikai rikai
+# Build the docker image
+docker build --tag rikai --network host .
+# Run the image
+docker run -p 0.0.0.0:8888:8888/tcp rikai:latest jupyter lab -ip 0.0.0.0 --port 8888
+```
+
+If successful, the console should then print out a clickable link to JupyterLab. You can also
+open a browser tab and go to `localhost:8888`.
 
-1. First build the rikai jar
+### <a name="Extras"></a>Install from pypi
 
-``` bash
-# from rikai root
-sbt publishLocal # assumes Scala 2.12 and version is 0.0.1
-```
+Base rikai library can be installed with just `pip install rikai`. Dependencies for supporting
+pytorch (pytorch and torchvision), aws (boto), jupyter (matplotlib and jupyterlab) are all part of 
+optional extras. Many open-source datasets also use Youtube videos so we've also added pafy and 
+youtube-dl as optional extras as well. 
 
-``` bash
-# from rikai root
-# if the base system is linux and you want to expose the gpu, ensure `nvidia-docker2` is installed first
-docker build --tag rikai --network host .
-```
+For example, if you want to use pytorch in Jupyter to train models on rikai datasets in s3 
+containing Youtube videos you would run:
+
+`pip install rikai[pytorch,aws,jupyter,youtube]`
+
+If you're not sure what you need and don't mind installing some extra dependencies, you can 
+simply install everything:
 
-2. Running the image
+`pip install rikai[all]`
 
-``` bash
-docker run -p 0.0.0.0:8888:8888/tcp rikai:latest jupyter lab --ip 0.0.0.0 --port 8888 --NotebookApp.quit_button=True --NotebookApp.custom_display_url=http://127.0.0.1:8888
+### <a name="Source"></a>Install from source
+
+To build from source you'll need python as well as Scala with sbt installed:
+
+```bash
+# Clone the repo
+git clone git@github.com:eto-ai/rikai rikai
+# Build the jar
+sbt publishLocal
+# Install python package
+cd python
+pip install -e . # pip install -e .[all] to install all optional extras (see "Install from pypi")
 ```
 
-The console should print out a clickable link to JupyterLab. Run the provided notebook to validate that spark standalone, pytorch, and rikai are installed correctly.
+## <a name="SparkSetup"></a>Local Spark Setup
+
+If you're running Spark locally, you'll need to add the rikai jar when creating the Spark session.
+If you want to read/write data from/to S3, you'll need to add additional options as well.
+
+Add appropriate options when creating the SparkSession:
+
+```python
+spark = (
+   SparkSession
+      .builder
+      .appName('rikai')          
+      .config('spark.jars.packages', 'ai.eto.rikai:rikai-core:0.0.1')
```

Comment:
are we going to change this to `0.0.1-SNAPSHOT`?

Reply:
Should just do 0.0.1 for the release?

Reply:
oh you want to change this to -SNAPSHOT leading up the release? yeah that makes sense. I was planning to have a separate repo to take care of the mvn -SNAPSHOT release